src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -135,6 +218,28 @@ public string TranslateToHostPath(string path)
 
             return path;
         }
+
+        public void AddPortMappings(List<PortMapping> portMappings)
+        {
+            foreach (var port in portMappings)
+            {
+                PortMappings.Add(port);
+            }
+        }
+
+        public void ExpandProperties(Variables variables)
+        {
+            // Expand port mapping
+            variables.ExpandValues(UserPortMappings);
```

Comment:
why not do similar thing to port, just like volume?

Reply:
Once volumes are expanded, they are done. Ports get expanded to replace variables in the string, but still have to be sent to Docker before getting the actual mappings back at runtime. So they cant be set yet